src/lightning_app/core/api.py

```diff
@@ -234,6 +236,29 @@ async def post_state(
     api_app_delta_queue.put(DeltaRequest(delta=Delta(deep_diff)))
 
 
+@fastapi_service.put("/api/v1/upload_file/{filename}")
+async def upload_file(filename: Optional[str] = None, uploaded_file: UploadFile = File(...)):
+    try:
+        filename = filename if filename else uploaded_file.filename
+        with TemporaryDirectory() as tmp:
+            drive = Drive(
+                "lit://uploaded_files",
+                component_name="file_server",
+                allow_duplicates=True,
+                root_folder=tmp,
+            )
+            tmp_file = os.path.join(tmp, filename)
+
+            with open(tmp_file, "wb") as f:
+                while content := await uploaded_file.read(10240):
+                    f.write(content)
+
+            drive.put(filename)
+    finally:
+        uploaded_file.file.close()
```

Comment:
Here we are closing the file, but where do we open it? What if an unopened file gets passed in?

Reply:
I think this is a FastAPI requirement.

Reply:
Removed.